src/Output/StrictUnifiedDiffOutputBuilder.php

```diff
@@ -0,0 +1,325 @@
+                continue;
+            }
+
+            $this->changed = true;
+
+            if (false === $hunkCapture) {
+                $hunkCapture = $i;
+            }
+
+            if (1 === $entry[1]) { // added
+                ++$toRange;
+            }
+
+            if (2 === $entry[1]) { // removed
+                ++$fromRange;
+            }
+        }
+
+        if (false === $hunkCapture) {
+            return;
+        }
+
+        // we end here when cutoff (commonLineThreshold) was not reached, but we where capturing a hunk,
+        // do not render hunk till end automatically because the number of context lines might be less than the commonLineThreshold
+
+        $contextStartOffset = $hunkCapture - $this->contextLines < 0
+            ? $hunkCapture
+            : $this->contextLines
+        ;
+
+        // prevent trying to write out more common lines than there are in the diff _and_
+        // do not write more than configured through the context lines
+        $contextEndOffset = \min($sameCount, $this->contextLines);
+
+        $fromRange -= $sameCount;
+        $toRange -= $sameCount;
+
+        $this->writeHunk(
+            $diff,
+            $hunkCapture - $contextStartOffset,
+            $i - $sameCount + $contextEndOffset + 1,
+            $fromStart - $contextStartOffset,
+            $fromRange + $contextStartOffset + $contextEndOffset,
+            $toStart - $contextStartOffset,
+            $toRange + $contextStartOffset + $contextEndOffset,
+            $output
+        );
+    }
+
+    private function writeHunk(
+        array $diff,
+        int $diffStartIndex,
+        int $diffEndIndex,
+        int $fromStart,
+        int $fromRange,
+        int $toStart,
+        int $toRange,
+        $output
+    ) {
+        \fwrite($output, '@@ -' . $fromStart);
+
+        if (!$this->collapseRanges || 1 !== $fromRange) {
+            \fwrite($output, ',' . $fromRange);
+        }
+
+        \fwrite($output, ' +' . $toStart);
+        if (!$this->collapseRanges || 1 !== $toRange) {
+            \fwrite($output, ',' . $toRange);
+        }
+
+        \fwrite($output, " @@\n");
+
+        for ($i = $diffStartIndex; $i < $diffEndIndex; ++$i) {
+            if ($diff[$i][1] === 1) { // added
+                $this->changed = true;
+                \fwrite($output, '+' . $diff[$i][0]);
+            } elseif ($diff[$i][1] === 2) { // removed
+                $this->changed = true;
+                \fwrite($output, '-' . $diff[$i][0]);
+            } elseif ($diff[$i][1] === 0) { // same
+                \fwrite($output, ' ' . $diff[$i][0]);
+            } elseif ($diff[$i][1] === self::$noNewlineAtOEFid) {
+                $this->changed = true;
+                \fwrite($output, $diff[$i][0]);
+            }
+            //} elseif ($diff[$i][1] === 3) { // custom comment inserted by PHPUnit/diff package
+                //  skip
+            //} else {
+                //  unknown/invalid
+            //}
```

Comment:
WIP or shall be dropped ?

Reply:
it is a kind of documentation;
I kept on rethinking what possible other cases there could be and why these are handled (because there is no need to handle those)